src/asset/unit.ts

```diff
@@ -5,25 +5,33 @@ import {getAddress} from "../rpc/getAddress";
 import {executeAssetOperation} from "./executeAssetOperation";
 import formatBalance from "@polkadot/util/format/formatBalance";
 import {Balance} from "@polkadot/types/interfaces";
+import {Configuration} from "../configuration/configuration";
 
-export function getPolkadotAssetDescription(balance: number|string|Balance, address: string): Asset {
+export function getPolkadotAssetDescription(
+  balance: number|string|Balance,
+  address: string,
+  configuration: Configuration): Asset
+{
+  // eslint-disable-next-line max-len
+  const customViewUrl = configuration.unit.customViewUrl ? configuration.unit.customViewUrl.replace("${address}", address) : "";
+  const image = configuration.unit.image ? configuration.unit.image : "";
   return {
-    balance: formatBalance(balance, {decimals: 12, withSi: true, withUnit: false}),
-    customViewUrl: `https://polkascan.io/pre/kusama/account/${address}`,
-    decimals: 0,
+    balance: formatBalance(balance, {decimals: configuration.unit.decimals, withSi: true, withUnit: false}),
+    customViewUrl: customViewUrl,
+    decimals: configuration.unit.decimals,
```

Comment:
I think this should be 0 or else Metamask will try to format balance. Wonder if it would be better to allow Metamask to format?
Can you try to pass raw balance string and decimals and see how it's gonna end up?